pkg/kubelet/kubelet.go

```diff
@@ -232,12 +232,13 @@ func (kl *Kubelet) GarbageCollectLoop() {
 }
 
 func (kl *Kubelet) getUnusedImages() ([]string, error) {
-	kl.pullLock.Lock()
-	defer kl.pullLock.Unlock()
 	return dockertools.GetUnusedImages(kl.dockerClient)
 }
 
 func (kl *Kubelet) GarbageCollectImages() error {
+	kl.pullLock.Lock()
```

Comment:
I agree.  I think this is what caused the problem with the e2e tests.  I'd rather see us do an inspect on the image prior to delete and validate that it still has no tags.  Won't fix the potential race, but it will close it a whole bunch.